packages/cloudflare/src/api/cloudflare-context.ts

```diff
@@ -0,0 +1,175 @@
+ */
+const cloudflareContextSymbol = Symbol.for("__cloudflare-context__");
+
+/**
+ * Utility to get the current Cloudflare context
+ *
+ * @returns the cloudflare context
+ */
+export function getCloudflareContext<
+  CfProperties extends Record<string, unknown> = IncomingRequestCfProperties,
+  Context = ExecutionContext,
+>(): CloudflareContext<CfProperties, Context> {
+  const global = getInternalGlobalThis<CfProperties, Context>();
+
+  const cloudflareContext = global[cloudflareContextSymbol];
+
+  if (!cloudflareContext) {
+    // the cloudflare context is initialized by the worker and is always present in production/preview
+    // during local development (`next dev`) it might be missing only if the developers hasn't called
+    // the `initOpenNextCloudflareForDev` function in their Next.js config file
+    const getContextFunctionName = getCloudflareContext.name;
+    const initFunctionName = initOpenNextCloudflareForDev.name;
+    throw new Error(
+      `\n\n\`${getContextFunctionName}\` has been called during development without having used` +
+        ` the \`${initFunctionName}\` function inside the Next.js config file.\n\n` +
+        `In order to use \`${getContextFunctionName}\` import and call ${initFunctionName} in the Next.js config file.\n\n` +
+        "Example: \n   ```\n   // next.config.mjs\n\n" +
+        `   import { ${initFunctionName} } from "@opennextjs/cloudflare";\n\n` +
+        `   ${initFunctionName}();\n\n` +
+        "   /** @type {import('next').NextConfig} */\n" +
+        "   const nextConfig = {};\n" +
+        "   export default nextConfig;\n" +
+        "   ```\n" +
+        "\n(note: currently middlewares in Next.js are always run using the edge runtime)\n\n"
+    );
+  }
+
+  return cloudflareContext;
+}
+
+/**
+ * Performs some initial setup to integrate as best as possible the local Next.js dev server (run via `next dev`)
+ * with the open-next Cloudflare adapter
+ *
+ * (Currently all the setup that this function performs is making sure that `getCloudflareContext` works as intended,
+ * in the future mode improvements might be added)
+ *
+ * Note: this function should only be called inside the Next.js config file
+ */
+export async function initOpenNextCloudflareForDev() {
+  const context = await getCloudflareContextFromWrangler();
+
+  addCloudflareContextToNodejsGlobal(context);
+
+  await monkeyPatchVmModuleEdgeContext(context);
+}
+
+/**
+ * Adds the cloudflare context to the global scope in which the Next.js dev node.js process runs in, enabling
+ * future calls to `getCloudflareContext` to retrieve and return such context
+ *
+ * @param cloudflareContext the cloudflare context to add to the node.sj global scope
+ */
+function addCloudflareContextToNodejsGlobal(cloudflareContext: CloudflareContext<CfProperties, Context>) {
+  const global = getInternalGlobalThis<CfProperties, Context>();
+  global[cloudflareContextSymbol] = cloudflareContext;
```

Comment:
When I see `getInternalGlobalThis(...)` I feel like the function is "doing" something. It is actually a noop and its only purpose it to cast. Can we just directly:

```suggestion
  const global = globalThis as InternalGlobalThis<CfProperties, Context>;
  global[cloudflareContextSymbol] = cloudflareContext;
```

_(global can even be inlined here)_

Reply:
yes I was thinking that maybe `getInternalGlobalThis` might be too much as well, I'm happy to remove it 👍

Reply:
done 👍